client/cmd/dexc/main.go

```diff
@@ -25,22 +25,6 @@ var log slog.Logger
 
 func main() {
 	appCtx, cancel := context.WithCancel(context.Background())
-	// Catch ctrl+c. This will need to be smarter eventually, probably displaying
-	// a modal dialog to confirm closing, especially if servers are running or if
-	// swaps are in negotiation.
-	killChan := make(chan os.Signal, 1)
-	signal.Notify(killChan, os.Interrupt)
-	var clientCore *core.Core
-	var shutdown func()
-	shutdown = func() {
-		<-killChan
-		if clientCore == nil || clientCore.PromptShutdown(killChan) {
-			cancel()
-			return
-		}
-		go shutdown()
-	}
-	go shutdown()
```

Comment:
`shutdown` is started in a goroutine so there is was a race for `clientCore` to be assigned below at `core.New` and the read inside `shutdown`.